dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java

```diff
@@ -663,31 +661,31 @@ public String getUser() {
     return getOptions().valueOf(optionUser);
   }
 
-  // -password has optional argument, and if not provied
-  // should be asked from command line
+  @Nonnull
+  public String getUserOrFail() {
+    String user = getOptions().valueOf(optionUser);
+    if (user == null) {
+      throw new MetadataDumperUsageException("Required username was not provided.");
+    }
+    return user;
+  }
+
   @CheckForNull
-  public String getPassword() {
-    if (!getOptions().has(optionPass)) {
+  public String getPasswordIfProvided() {
+    if (getOptions().has(optionPass)) {
+      return getPassword();
+    } else {
       return null;
     }
-    String pass = getOptions().valueOf(optionPass);
-    if (pass != null) {
-      return pass;
-    }
-    // Else need to ask & save.
-    if (askedPassword != null) {
-      return askedPassword;
-    }
+  }
 
-    Console console = System.console();
-    if (console == null) {
-      LOG.info("Cannot prompt for password, Console not available");
-      return null;
+  @Nonnull
+  public String getPassword() {
```

Comment:
I find the naming of `getPassword` and `getPasswordIfProvided` a bit confusing. I would suggest renaming this method to `getPasswordOrPrompt`.

Alternatively, consider restructuring the code to have only a single `getPassword` method and to make the flag behavior more explicit, for example:
```
/**
 * Behavior of the dumper based on the state of the {@code --password} flag.
 */
enum PasswordPromptBehavior {
  /**
   * <p>If the flag is not provided, then empty {@code Optional} is returned.
   * <p>If the flag is provided without the parameter, then the user is prompted to enter the password interactively.
   * <p>If the flag is provided with the parameter, then the password from the parameter is used.
   */
  PROMPT_ONLY_IF_FLAG_PROVIDED,

  /**
   * <p>If the flag is provided with the parameter, then the password from the parameter is used.
   * <p>Otherwise (i.e. if the flag is not provided or if it is provided but without the parameter) the user is prompted to enter the password interactively.
   */
  PROMPT_UNLESS_PARAMETER_PROVIDED
}
public Optional<String> getPassword(PasswordPromptBehavior behavior) { ... }
```

Reply:
Done, I chose the first option.

Having two methods is more convenient in cases where we don't need to handle null passwords, because wrapping the result in an `Optional` would be redundant.